spec/granite/transactions/destroy_spec.cr

```diff
@@ -55,5 +55,37 @@ module {{adapter.capitalize.id}}
       end
     end
   end
+
+  class ParentWithCallback < Granite::Base
+    adapter {{ adapter.id }}
+    table_name parents
+
+    before_destroy :cancel_destroy
+
+    private def cancel_destroy
+      abort!("Canceling destroy")
+    end
+  end
+
+  describe "{{ adapter.id }} #destroy!" do
+    it "destroys an object" do
+      parent = Parent.new
+      parent.name = "Test Parent"
+      parent.save
+
+      id = parent.id
+      parent.destroy
+      found = Parent.find id
+      found.should be_nil
+    end
+
+    it "does not destroy an invalid object but raise an exception" do
```

Comment:
I agree on that but i could not find an way to raise an database exception to verify this behaviour. Granite will generate the following query "DELETE FROM "parents" WHERE "id"=$1". If the record does not exist it will still run the query and succeed. So i could not raise an "DB::Error" so i was verify the behaviour with an "Granite::Callbacks::Abort" error. I verified that the delete query never runned. I will add an test that verifies that the record is still there

Reply:
Great explanation. To be clear, I don't have any problem with the way you're triggering the error condition, I was just commenting on the description text of the test itself. :]